### PR TITLE
Windows support and other various changes

### LINKED ISSUE #9: Incorporate @compnerd's patch for Windows support
It would be great if we could incorporate https://github.com/marmelroy/Zip/pull/246/files, which would enable this to work on Windows. Today there's no well-maintained ZIP implementation that includes Windows support, and you're achingly close!

### PR DESCRIPTION
**Pull request by fpseverino**
### Issue #9 

Originally authored by @compnerd in https://github.com/marmelroy/Zip/pull/246

- Add support for Windows
- Add Windows CI tests
- Drop support for Swift 5.8
- Adopt `swift-format`
- Add CI for iOS and MUSL
- Various bug fixes

### CODE REVIEW

File: Sources/Zip/ArchiveFile.swift [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import Foundation
```
Comment (ptoffy) [inline Sources/Zip/ArchiveFile.swift]: I wonder if we could start using `FoundationEssentials` here

File: Sources/Zip/Zip.swift [hunk 2]
```diff
@@ -92,40 +89,44 @@ public class Zip {
...
             }
 
-            let creationDate = Date()
             let directoryAttributes: [FileAttributeKey: Any]?
-            #if os(Linux) && swift(<6.0)
-                // On Linux, setting attributes is not yet really implemented.
-                // In Swift 4.2, the only settable attribute is `.posixPermissions`.
-                // See https://github.com/apple/swift-corelibs-foundation/blob/swift-4.2-branch/Foundation/FileManager.swift#L182-L196
+            #if (os(Linux) || os(Windows)) && swift(<6.0)
+                // On Linux and Windows, setting attributes is not yet really implemented.
```
Comment (ptoffy) [inline Sources/Zip/Zip.swift]: I'm not an expert but is this a problem? If so we should at least add a warning for people not to expect this to work

File: Sources/Zip/ArchiveFile.swift (hunk 1 above)
Comment (fpseverino) [reply Sources/Zip/ArchiveFile.swift]: I have to check `#if swift(>=6.0)` to `import` it and use the old `Foundation` otherwise, right?

Comment (ptoffy) [reply Sources/Zip/ArchiveFile.swift]: I think it should just be available. I mean if you're not using Swift 6 locally yourself and it works it should be fine

Comment (fpseverino) [reply Sources/Zip/ArchiveFile.swift]: I'm on Swift 5.10 and it doesn't work. Seems to only work in Swift 6, I implemented it [like this](https://github.com/vapor-community/Zip/pull/10/commits/5ecd26474ccb9c228e13edf6cd8c08f5ec3c3486).

File: Sources/Zip/Zip.swift (hunk 2 above)
Comment (fpseverino) [reply Sources/Zip/Zip.swift]: I'm not an expert either, it was always like that in the original repo. In Swift 6 they can be set on all platforms, but looking at [the docs](https://developer.apple.com/documentation/foundation/fileattributekey/1418187-creationdate) right now I discovered that we may need to add a privacy manifest if we set any attribute outside of `.posixPermissions`

**Comment by fpseverino** [comment]
I just recently discovered [`swift-system`](https://swiftpackageindex.com/apple/swift-system), which solves most of the problems and inconsistencies with Windows, but requires some work to integrate it.

Since I won't have a lot of time in the near future, and since the package seems to work fine on Windows at the moment, I'll postpone the switch to `swift-system` until the next PR.

File: Sources/Zip/ArchiveFile.swift (hunk 1 above)
Comment (fpseverino) [reply Sources/Zip/ArchiveFile.swift]: I've discovered the proper way to include FoundationEssentials and did that in any file where we don't need the whole Foundation

File: Sources/Zip/Zip.swift (hunk 2 above)
Comment (fpseverino) [reply Sources/Zip/Zip.swift]: I've decided to just remove the `.creationDate` and `.modificationDate` attributes on all platforms. Doesn't seem to be a problem, ZIPFoundation doesn't use them either

Comment (fpseverino) [reply Sources/Zip/Zip.swift]: ZIPFoundation sets them only on Apple platforms, now they are available in `swift-foundation` on all platforms so for Swift 6.0+ we set them on Linux and Windows too. Doesn't seem to be a problem. Regarding the privacy manifest, this a package oriented more on the server-side, but if anyone wants to open a PR to add it we can add it

**Review by compnerd — COMMENTED** [review]
There are a lot of superfluous changes that make this difficult to review. Can you split out the changes to the doc comments into a separate change as a starting point?

**Comment by fpseverino** [comment]
@compnerd I made [a new PR](https://github.com/vapor-community/Zip/pull/12) with the changes related to Windows (in Zip.swift there are stil some unrelated changes that are hard to revert, I'm sorry)

**Review by gwynne — APPROVED** [review]

**Pull request merged**